tests/components/template/test_light.py

```diff
@@ -521,18 +469,13 @@ def test_white_value_action_no_template(self):
                             "turn_on": {
                                 "service": "light.turn_on",
                                 "entity_id": "light.test_state",
+                                "data": {"white_value": "{{white_value}}"},
                             },
                             "turn_off": {
                                 "service": "light.turn_off",
                                 "entity_id": "light.test_state",
                             },
-                            "set_white_value": {
-                                "service": "test.automation",
```

Comment:
I believe this PR breaks the option to use custom services which is why this test broke

Reply:
> I believe this PR breaks the option to use custom services which is why this test broke.

It doesn't break the option to, I am still working on the tests 😄. Unless I am misunderstanding, I don't see a limitation that prevents a user from specifying a custom script on the `turn_on` script.

I think I can fix these tests—let me try again.

> I think the idea of setting all values in one call is fine but the custom script support should stay. Perhaps revert the boolean-approach back to scripts and make this PR about optionally applying all scripts in the turn_on action?
>
> It would prevent the PR from being a breaking change and it would still solve #34915

I think this would also be confusing to document, especially for new users. It would not be clear which method a user should use and would still allow for a user to run into #34915 without knowing/understanding why.

I agree it is an unfortunate breaking change, but it seems (solely based on observing previous decisions of the project) that preventing possibility of error/bug/unintentional output is preferred even if it incurs a breaking change.

Reply:
Resolving since I am passing all tests and no functionality has been lost